File: src/my_cpp_pkg/src/Addnum_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using std::placeholders:: _1;
using std::placeholders:: _2;

class AddNumNode : public rclcpp::Node 
{
public:
    AddNumNode() : Node("AddNum") 
    {
      server = this->create_service<example_interfaces::srv::AddTwoInts>("AddNum",
      std::bind(&AddNumNode::callbacksum,this,_1,_2));
       RCLCPP_INFO(this->get_logger(),"server is online");
    }

private:
    void callbacksum(const example_interfaces::srv::AddTwoInts::Request::SharedPtr request,
            const example_interfaces::srv::AddTwoInts::Response::SharedPtr response)
            {
                 response->sum = request->a + request->b;
                 RCLCPP_INFO(this->get_logger(), "%ld + %ld = %ld", 
                            request->a,request->b,response->sum);
            }
    rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<AddNumNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/Addnum_server.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class AddNumNode(Node): 
    def __init__(self):
        super().__init__("AddNum") 
        self.server = self.create_service(AddTwoInts,"AddNum",self.callbacksum)
        self.get_logger().info("service has started")

    def callbacksum(self,request,response):
        response.sum = request.a +request.b
        self.get_logger().info(f"{request.a} + {request.b} = {response.sum}")
        return response

def main(args=None):
    rclpy.init(args=args)
    node = AddNumNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()